algorithmic-strategies/divide-and-conquer.md

```diff
@@ -0,0 +1,292 @@
+
+<br />
+
+<details>
+
+<summary>We can view a recursive implementation of binary search by expanding this section.</summary>
+
+```python
+def recursive_binary_search(array, to_find, low=0, high=None):
+    if high is None:
+        high = len(array)
+
+    if high <= low:
+        return None
+
+    mid = (high + low) // 2
+
+    if array[mid] == to_find:
+        return mid
+    elif array[mid] > to_find:
+        high = mid
+    else:
+        low = mid + 1
+
+    return recursive_binary_search(array, to_find, low, high)
+```
+
+</details>
+
+## Example: QuickSort
+
+QuickSort is an algorithm which takes a divide-and-conquer approach to sorting an array by using the following steps:
+
+1. If the array is only one element or empty, we are done, the array is sorted.
+1. Pick an element from the array as the _pivot_.
+1. Move all elements smaller than the pivot to the left and all elements larger than the pivot to the right. Note that the pivot is now in the correct index. Picking a pivot and rearranging the elements is referred to as _partitioning_.
+1. Perform QuickSort on the left and right sides of the pivot.
+
+![QuickSorting the array (7, 3, 9, 1, 6, 8, 2, 5) where the pivot is chosen by taking the last element in the array. The first pivot is 5, which after rearranging gives the array (3, 1, 2, 5, 6, 8, 9, 7). The left array (3, 1, 2) and right array (6, 8, 9, 7) are QuickSorted. 2 is selected as the pivot, and after rearranging, the left array becomes (1, 2, 3). The left and right arrays are (1) and (3), which are both 1 element, meaning they are sorted. Returning to (6, 8, 9, 7), 7 is picked as the pivot. After rearranging, the array becomes (6, 7, 9, 8). The left array is (6), which is sorted. The right array is (9, 8). 8 is selected as the pivot. After rearranging, the array becomes (8, 9). There is no left array, and the right is (9), which is sorted. All subarrays have been sorted, meaning the whole array is sorted, having become (1, 2, 3, 5, 6, 7, 8, 9).](../assets/algorithmic-strategies_divide-and-conquer_quick-sort.png)  
+_Fig. Tracing through an application of QuickSort in which the final element in a subarray is chosen as the pivot._
+
+In terms of divide and conquer, we pick a pivot and move smaller elements to the left, and larger elements to the right. This leaves us with two smaller subproblems: sorting the elements on the left, and those on the right. We call QuickSort on each side.
+
+If the pivot is relatively well chosen we will divide the list _log n_ levels deep, and shift at most _n_ elements with each level of division, arriving at a _O(n log n)_ runtime.
+
+**Weakness of QuickSort**: The Pivot
+
+However, notice we said **if** the pivot is well chosen. If the pivot does not break the list into two relatively equal subarrays it will not arrive at a _O(n log n)_ runtime. Instead it will approach a _O(n<sup>2</sup>)_ runtime.
```

Comment:
Transition into this new section. Added a few commas for fun.

```suggestion
Again, QuickSort is _O(n log n)_ **if** the pivot is well-chosen. If the pivot does not break the list into two relatively equal subarrays, it will not arrive at a _O(n log n)_ runtime. Instead, it will approach a _O(n<sup>2</sup>)_ runtime.
```

Reply:
Thanks for rephrasing the beginning. I didn't really like the exclusive "we" (the instructional staff) as opposed to our more typical inclusive we.